sklearn/impute/_knn.py

```diff
@@ -0,0 +1,271 @@
+import numpy as np
+
+from ..base import BaseEstimator, TransformerMixin
+from ..utils.validation import FLOAT_DTYPES
+from ..metrics import pairwise_distances
+from ..metrics.pairwise import _NAN_METRICS
+from ..neighbors.base import _get_weights
+from ..neighbors.base import _check_weights
+from ..utils import check_array
+from ..utils import is_scalar_nan
+from ..utils.mask import _get_mask
+from ..utils.validation import check_is_fitted
+
+
+class KNNImputer(TransformerMixin, BaseEstimator):
+    """Imputation for completing missing values using k-Nearest Neighbors.
+
+    Each sample's missing values are imputed using values from `n_neighbors`
```

Comment:
"using values" -> "using the mean value". I'd be happy to remove all sentences after this one. Indeed, if there were to be another sentence, I'd choose to mention how distance is calculated, basically "Two samples are close if the features that neither is missing are close."